### ISSUE TITLE
Overhaul the reactnative code

### ISSUE THREAD

**Issue by andygrillo** [openeew/openeew-app-reactnative#13]
This repo seems to have a better functioning notification system for iOS/Android:
https://github.com/zo0r/react-native-push-notification

The code we have now does not successfully sound in every notification event. Perhaps we could create a basic app based on zo0r's repo which simply received notifications ?

**Comment by krook** [comment]
Push notifications on React Native have caused issues in other projects I've seen. 

I wonder if it makes sense to also explore native apps and underlying SDKs which might be more reliable.

**Comment by andygrillo** [comment]
Right, that makes sense then. What about Flutter?

**Comment by krook** [comment]
I have not used that. I was thinking if we had Java/Kotlin apps for Android and Swift apps for iPhone.

**Comment by andygrillo** [comment]
That would be great. If someone however could achieve both platforms with a single code base it would have benefits. But I defer to you on this!

**Comment by marcelorubim** [comment]
Just change the application to use react-native-push-notification should fix the notification system, right? 

I'm gonna help with this.

**Comment by andygrillo** [comment]
Amazing, please have a go :) 

My only suggestion may be to start a clean project and use the existing code as a guide (loosely). The existing code relies on the FCM system from Grillo and I believe it also reads historic earthquakes from a firebase DB also from Grillo. 

I wonder if the historic earthquakes could instead be read from the postgres database we have established in the detection docker repo ?

Not sure if you have alternative suggestions to Firebase messaging so that people are not tied to google...

**Comment by marcelorubim** [comment]
Nice!

I like the idea to start a clean project. I had a bit of a hard time to make this run.

I think we would be needing a backing end to read the data from the Postgres or do you already have it?

**Comment by johnwalicki** [comment]
If you are displaying historic earthquakes, i would not duplicate the data. Retrieve the dataset directly from the S3 bucket.

**Comment by andygrillo** [comment]
Agreed although we need to decide if the app shows events from this users
network, in which case it should read from the earthquake table in the
docker database, and potentially info about each sensor from the device
table.

In addition it could be great to read universal OpenEEW data from the S3...

**Comment by marcelorubim** [comment]
I'm going to start working on this in a clean project. :)

You can assign this to me.

**Comment by andygrillo** [comment]
Thanks @marcelorubim ! if you haven't already we would love to have you in the #apps channel on slack. That way you can let us know as anything comes up.

**Comment by mghildiy** [comment]
I would like to contribute here.

**Comment by andygrillo** [comment]
Thanks @mghildiy :) perhaps we can discuss this also in slack #apps channel ? I am not sure how far @marcelorubim has got to.

**Comment by andygrillo** [comment]
One thought @mghildiy and @marcelorubim is to integrate the OpenEEW sensor provisioning feature into this app (so that its not just an alert app).

This is something we discussed in last week's townhall. it would mean that a user can quickly pass over wifi and location info to the sensor when setting it up.

As we use ESP32 devices, we could consider the Smart Config protocol (eg https://apps.apple.com/ec/app/grillo/id1478536152). There is a React Native library that does this: 
https://www.npmjs.com/package/react-native-smartconfig-2

Could someone please look at this library and see if its possible to pass over not just Wifi credentials, but other data (eg the phone's latitutde and longitude) ?

**Comment by marcelorubim** [comment]
Well, I've done some part of the app. It's far for be functional but did some screens with i18n support (currently Spanish and English).

I've commit at my GitHub so anyone can take a look:

https://github.com/marcelorubim/Grillo

Latter I'll do the merging with this repo fork.
